File: LibraryBackend/UI/elibrary_mobile/lib/models/book_exchange.dart
import 'package:json_annotation/json_annotation.dart';
import 'book.dart';
import 'user.dart';
import 'book_exchange_status.dart'; 

part 'book_exchange.g.dart';

@JsonSerializable()
class BookExchange {
  int? id;

  int? offerUserId;
  User? offerUser;

  int? receiverUserId;
  User? receiverUser;

  int? offerBookId;
  Book? offerBook;

  int? receiverBookId;
  Book? receiverBook;

  bool? offerUserAction;
  bool? receiverUserAction;

  BookExchangeStatus? bookExchangeStatus;

  BookExchange({
    this.id,
    this.offerUserId,
    this.offerUser,
    this.receiverUserId,
    this.receiverUser,
    this.offerBookId,
    this.offerBook,
    this.receiverBookId,
    this.receiverBook,
    this.offerUserAction = false,
    this.receiverUserAction = false,
    this.bookExchangeStatus = BookExchangeStatus.BookDeliveryPhase,
  });

  factory BookExchange.fromJson(Map<String, dynamic> json) =>
      _$BookExchangeFromJson(json);

  Map<String, dynamic> toJson() => _$BookExchangeToJson(this);
}

File: LibraryBackend/Library.Models/Entities/BookExchange.cs
using Library.Models.Enums;
using Library.Services.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Library.Models.Entities
{
    public class BookExchange
    {
        public int Id { get; set; }

        public int OfferUserId { get; set; }
        public User OfferUser { get; set; }

        public int ReceiverUserId { get; set; }
        public User ReceiverUser { get; set; }

        public int OfferBookId { get; set; }
        public Book OfferBook { get; set; }

        public int ReceiverBookId { get; set; }
        public Book ReceiverBook { get; set; }

        public bool OfferUserAction { get; set; } = false;
        public bool ReceiverUserAction { get; set; } = false;
        public BookExchangeStatus BookExchangeStatus { get; set; } = BookExchangeStatus.PendingApproval;

    }
}
